lib/src/kbasesearchengine/parse/KeywordParser.java

```diff
@@ -33,6 +35,20 @@ public static ParsedObject extractKeywords(
             final ObjectLookupProvider lookup,
             final List<GUID> objectRefPath)
             throws IOException, ObjectParseException, IndexingException, InterruptedException {
+
+        // check pre-conditons
+        Utils.notNullOrEmpty(json, "json is a required parameter");
+
+        Utils.nonNull(indexingRules, "indexingRules is a required parameter");
+
+        for (IndexingRules rule: indexingRules) {
+            try {
+                rule.validate();
```

Comment:
is this still needed given the validate in the objectparsingrules?

Reply:
Until I can refactor to make an immutable class with builder and there is only one way to instantiate. Currently an instance can be created and morphed in many ways.